app/policies/csv_download_policy.rb

```diff
@@ -0,0 +1,12 @@
+class CsvDownloadPolicy
+  attr_reader :current_user, :csv_download
+
+  def initialize(current_user, csv_download)
+    @current_user = current_user
+    @csv_download = csv_download
+  end
+
+  def download?
+    @current_user == @csv_download.user || @current_user.support? || @current_user.organisation == @csv_download.organisation
+  end
```

Comment:
What do you think about also allowing users from affiliated organisations (to the download organisation) to also be able to download the link? E.g. if a managing agent shares the link with a stock owner.

Reply:
I feel like those organisations might have access to different org logs because they have different relationships, so it might potentially expose logs that wouldn't otherwise be visible